src/test/kotlin/io/spine/text/TextFactorySpec.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2022, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.text
+
+import io.kotest.matchers.collections.shouldContainInOrder
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+import io.spine.string.Separator
+import io.spine.testing.UtilityClassTest
+import io.spine.text.TextFactory.createText
+import io.spine.text.TextFactory.newLine
+import io.spine.text.TextFactory.text
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+
+@DisplayName("`TextFactory` should")
+internal class TextFactorySpec: UtilityClassTest<TextFactory>(TextFactory::class.java) {
+
+    private val nl = Separator.NL
+
+    companion object {
+        val linesWithSeparators: Array<String> = arrayOf(
+            " Fiz ${Separator.LF} buz? ",
+            " Foo ${Separator.CR} bar.",
+            "Ka ${Separator.CRLF} boom!"
+        )
+    }
+
+    @Nested
+    @DisplayName("prohibit line separators when building from many lines")
+    inner class Prohibition {
+
+        @Test
+        fun `with an array`() = assertThrowsOn {
+            text(linesWithSeparators)
+        }
+
+        @Test
+        fun `with an Iterable`() = assertThrowsOn {
```

Comment:
Let's settle on either single-quoting (`' ... '`) the type names in the names of tests, or not. Here it's a quote-free `Iterable`. Down below there's the one with quotes. 

I like quoting, personally. It reminds me of ol'—good days of using backticks in Java's `@DisplayName`s.